packages/authorization-updater/src/readModelService.ts

```diff
@@ -97,27 +97,37 @@ export function readModelServiceBuilder(
       return undefined;
     },
 
-    async getAgreement(
+    async getLatestAgreement(
       eserviceId: EServiceId,
       consumerId: TenantId
     ): Promise<Agreement | undefined> {
-      const data = await agreements.findOne(
-        {
-          "data.eserviceId": eserviceId,
-          "data.consumerId": consumerId,
-          "data.state": {
-            $in: [
-              agreementState.active,
-              agreementState.archived,
-              agreementState.suspended,
-            ],
+      const data = await agreements
+        .find(
```

Comment:
I think that we can safely keep `findOne` and delete `limit(1)` at the end with the sort you introduced